### ISSUE TITLE
zsh: username part of the prompt is empty

### ISSUE THREAD

**Issue by rqcy** [termux/termux-packages#424]
`%n` is an empty string in zsh's prompt, so:

```
PROMPT='%n@%M '
```

results in the following prompt: `@localhost`

Should be `u0_a65@localhost` or similar.

Reference: http://zsh.sourceforge.net/Doc/Release/Prompt-Expansion.html#Login-information

**Comment by igormp** [comment]
After fiddling a bit into zsh's source, I can't see anything apparently wrong with the functions to set the username (although I may be looking at the wrong place).

I checked the following codes:

>Starting at line 1082 @ Src/utils.c:
```c
char *
get_username(void)
{
#ifdef HAVE_GETPWUID
    struct passwd *pswd;
    uid_t current_uid;

    current_uid = getuid();
    if (current_uid != cached_uid) {
	cached_uid = current_uid;
	zsfree(cached_username);
	if ((pswd = getpwuid(current_uid)))
	    cached_username = ztrdup(pswd->pw_name);
	else
	    cached_username = ztrdup("");
    }
#else /* !HAVE_GETPWUID */
    cached_uid = getuid();
#endif /* !HAVE_GETPWUID */
    return cached_username;
}
```

>Starting at line 4368 @ Src/params.c:
```c
char *
usernamegetfn(UNUSED(Param pm))
{
    return get_username();
}

/* Function to set value of special parameter `USERNAME' */

/**/
void
usernamesetfn(UNUSED(Param pm), char *x)
{
#if defined(HAVE_SETUID) && defined(HAVE_GETPWNAM)
    struct passwd *pswd;

    if (x && (pswd = getpwnam(x)) && (pswd->pw_uid != cached_uid)) {
# ifdef USE_INITGROUPS
	initgroups(x, pswd->pw_gid);
# endif
	if (setgid(pswd->pw_gid))
	    zwarn("failed to change group ID: %e", errno);
	else if (setuid(pswd->pw_uid))
	    zwarn("failed to change user ID: %e", errno);
	else {
	    zsfree(cached_username);
	    cached_username = ztrdup(pswd->pw_name);
	    cached_uid = pswd->pw_uid;
	}
    }
#endif /* HAVE_SETUID && HAVE_GETPWNAM */
    zsfree(x);
}
```

The system's functions `getuid` and `getpwduid` seem to be working as expected, after following test I've run in my own phone:
```c
#include <unistd.h>
#include <sys/types.h>
#include <stdio.h>
#include <pwd.h>

int main(){

    typedef struct passwd {
        char   *pw_name;       /* username */
        char   *pw_passwd;     /* user password */
        uid_t   pw_uid;        /* user ID */
        gid_t   pw_gid;        /* group ID */
        char   *pw_gecos;      /* user information */
        char   *pw_dir;        /* home directory */
        char   *pw_shell;      /* shell program */
    }passwd;

    passwd *test;

    unsigned int id = getuid();
    test = getpwuid(id);

    printf("Id: %d \nUser: %s", id, test->pw_name);
    return 0;
}
```
Which returns, as expected:
```
Id: 10142 
User: u0_a142
```

If anyone has any idea as to where the empty var comes from, I'd be glad to help testing or in any other useful way.

Meanwhile, a little workaround to this would be to set the `$USERNAME` var with the value returned from `whoami`, since that's where the `%n` escape value gets its value from.

**Comment by fornwall** [comment]
@igormp Thanks for your investigation! It lead to the discovery of a fix, which has been applied to the `5.4.2-2` version of the zsh package. Update to it by running

    pkg up

**Comment by igormp** [comment]
Just to confirm that it indeed works now:
```
echo $USERNAME
u0_a142
```

So yey! I'm glad that I could help :smile:
